src/include/ipc/dai.h

```diff
@@ -61,30 +61,42 @@ enum sof_ipc_dai_type {
 	SOF_DAI_INTEL_ALH,		/**< Intel ALH */
 	SOF_DAI_IMX_SAI,                /**< i.MX SAI */
 	SOF_DAI_IMX_ESAI,               /**< i.MX ESAI */
+	SOF_DAI_MULTIDAI = 0xFFFF       /**< config with multiple elements of the same dai type */
 };
 
 /* general purpose DAI configuration */
 struct sof_ipc_dai_config {
 	struct sof_ipc_cmd_hdr hdr;
-	uint32_t type;		/**< DAI type - enum sof_ipc_dai_type */
+
+	/* FIXME endianness */
+	uint16_t type;		/**< DAI type - enum sof_ipc_dai_type */
+	uint16_t subdai_type;	/**< only used in multi-dai cases */
```

Comment:
I assume the comment is related to the structures being little-endian and that this could break backward compatibility if somehow it was big endian. But is it a real concern, did any of these structures ever was big endian?

We may establish that these should always be little endian for the future (and possibly have the endianness conversion macros ported to SOF?) so this isn't a problem. Not the subject of this PR though.

Reply:
At the moment we have no support for endianness differences between the host and DSP. That's probably something to fix for IPC2.0...

What I was indicating here was a reminder to myself to double-check I wouldn't break backwards compatibility on existing solutions.